shogun-lib/src/main/java/de/terrestris/shogun/lib/controller/security/permission/BasePermissionController.java

```diff
@@ -0,0 +1,837 @@
+            getGenericClassName(), entityId);
+
+        try {
+            Optional<S> entity = service.findOne(entityId);
+
+            if (entity.isEmpty()) {
+                throw getEntityNotFoundException(entityId);
+            }
+
+            groupInstancePermissionService.deleteAllFor(entity.get());
+
+            LOG.trace("Successfully deleted all group instance permissions for entity of type {} with ID {}",
+                getGenericClassName(), entityId);
+        } catch (AccessDeniedException ade) {
+            throw getAccessDeniedException(ade);
+        } catch (ResponseStatusException rse) {
+            throw rse;
+        } catch (Exception e) {
+            throw getDeleteException(e);
+        }
+    }
+
+    @DeleteMapping("/{id}/permissions/class/user")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void deleteUserClassPermissions(
+        @PathVariable("id") Long entityId
+    ) {
+        LOG.trace("Requested to delete all user class permissions for entity of type {} with ID {}",
+            getGenericClassName(), entityId);
+
+        try {
+            Optional<S> entity = service.findOne(entityId);
+
+            if (entity.isEmpty()) {
+                throw getEntityNotFoundException(entityId);
+            }
+
+            userClassPermissionService.deleteAllFor(entity.get());
+
+            LOG.trace("Successfully deleted all user class permissions for entity of type {} with ID {}",
+                getGenericClassName(), entityId);
+        } catch (AccessDeniedException ade) {
+            throw getAccessDeniedException(ade);
+        } catch (ResponseStatusException rse) {
+            throw rse;
+        } catch (Exception e) {
+            throw getDeleteException(e);
+        }
+    }
+
+    @DeleteMapping("/{id}/permissions/class/group")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void deleteGroupClassPermissions(
+        @PathVariable("id") Long entityId
+    ) {
+        LOG.trace("Requested to delete all group class permissions for entity of type {} with ID {}",
+            getGenericClassName(), entityId);
+
+        try {
+            Optional<S> entity = service.findOne(entityId);
+
+            if (entity.isEmpty()) {
+                throw getEntityNotFoundException(entityId);
+            }
+
+            groupClassPermissionService.deleteAllFor(entity.get());
+
+            LOG.trace("Successfully deleted all group instance permissions for entity of type {} with ID {}",
+                getGenericClassName(), entityId);
+        } catch (AccessDeniedException ade) {
+            throw getAccessDeniedException(ade);
+        } catch (ResponseStatusException rse) {
+            throw rse;
+        } catch (Exception e) {
+            throw getDeleteException(e);
+        }
+    }
+
+    protected ResponseStatusException getCreateException(Exception e) {
+        LOG.error("Error while setting permission: \n {}", e.getMessage());
+
+        throw getException(e);
```

Comment:
I generally find this exception handling somewhat confusing. You're logging the correct stack trace here, but throwing a new one here and in `getException` again will probably lead to some really weird stack frames (I can't even tell from looking at the code with how many causes we'll end up here). Maybe the situation can be clarified somewhat by just returning an exception here and in `getException` and throwing it once in the actual endpoint method.

That said, personally I find it much clearer to avoid exceptions when status codes are involved and instead use proper `HttpResponse` objects with an appropriate code.

Reply:
See 83db6c7 and bbcf211 for a simplified error handling using custom exceptions.

Reply:
LGTM, please merge.